pkg/resource/default.go

```diff
@@ -0,0 +1,121 @@
+
+// Error strings
+const (
+	errNoDefault        = "unable to locate a default resource class for claim kind"
+	errMultipleDefaults = "multiple default classes defined for claim kind"
+	errInvalidLabel     = "unable to set label for claim kind"
+)
+
+// DefaultReconciler reconciles resource claims to the
+// default resource class for their given kind. Predicates
+// ensure that only claims with no resource class reference
+// are reconciled.
+type DefaultReconciler struct {
+	client   client.Client
+	newClaim func() Claim
+}
+
+// NewDefaultReconciler creates a new DefaultReconciler for the claim kind
+func NewDefaultReconciler(m manager.Manager, of ClaimKind) *DefaultReconciler {
+	nc := func() Claim { return MustCreateObject(schema.GroupVersionKind(of), m.GetScheme()).(Claim) }
+
+	_ = nc()
+
+	r := &DefaultReconciler{
+		client:   m.GetClient(),
+		newClaim: nc,
+	}
+
+	return r
+}
+
+// Reconcile reconciles a claim to the default class reference for its kind
+func (r *DefaultReconciler) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	logDefault.V(logging.Debug).Info("Reconciling", "request", req)
+
+	ctx, cancel := context.WithTimeout(context.Background(), reconcileTimeout)
+	defer cancel()
+
+	claim := r.newClaim()
+	if err := r.client.Get(ctx, req.NamespacedName, claim); err != nil {
+		// There's no need to requeue if we no longer exist. Otherwise we'll be
+		// requeued implicitly because we return an error.
+		return reconcile.Result{}, errors.Wrap(IgnoreNotFound(err), errGetClaim)
+	}
+
+	// Get claim group and kind information
+	claimKind := claim.GetObjectKind()
+	claimGroupVersionKind := claimKind.GroupVersionKind()
+	claimGroupKind := claimGroupVersionKind.GroupKind()
+	claimGroupKindStr := strings.ToLower(claimGroupKind.String())
+
+	// Create label selector in crossplane-system namespace
+	options := client.InNamespace("crossplane-system")
+	if err := options.SetLabelSelector(claimGroupKindStr + "/default=" + "true"); err != nil {
+		return reconcile.Result{Requeue: false}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), errInvalidLabel)
+	}
+
+	// Get resource classes with claim kind as default
+	classes := &corev1alpha1.ResourceClassList{}
+	if err := r.client.List(ctx, options, classes); err != nil {
+		return reconcile.Result{Requeue: false}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), errNoDefault)
+	}
+
+	// Check to make sure there is only one default
+	// resource class defined
+	if len(classes.Items) != 1 {
+		return reconcile.Result{Requeue: false}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), errMultipleDefaults)
+	}
+
+	defaultClass := classes.Items[0]
+
+	defaultClassRef := &corev1.ObjectReference{
+		Name:      defaultClass.GetName(),
+		Namespace: defaultClass.GetNamespace(),
+	}
+
+	// Set class reference on claim to default resource lcass
+	claim.SetClassReference(defaultClassRef)
```

Comment:
```suggestion
	claim.SetClassReference(meta.ReferenceTo(classes.Items[0], corev1alpha1.ResourceClassGroupVersionKind)
```

The above will populate the UID, type metadata, etc, of the reference as well as the namespace and name.

Reply:
Sweet :)